tests/bluetooth/iso/bsim/main.c

```diff
@@ -0,0 +1,452 @@
+			LOG_ERR("Agument %u too long", i);
+			return -ENOMEM;
+		}
+
+		LOG_DBG("Adding %s argc is %d", arg, argc);
+
+		strcat(cmd_str, arg);
+		strcat(cmd_str, " ");
+	}
+
+	err = shell_execute_cmd(NULL, cmd_str);
+	if (err) {
+		LOG_ERR("Failed to send command %s", cmd_str);
+		return err;
+	}
+
+	va_end(argv);
+
+	return 0;
+}
+
+static int modules_configure(void)
+{
+	int ret;
+	char *arg_dev;
+	char *arg_command;
+	char *arg_value;
+
+	for (int i = 0; i < argc_copy; i = i + 3) {
+		/* Argument meant for tester framework */
+		if (strcmp(argv_copy[i], "tester") == 0) {
+			continue;
+		}
+
+		arg_dev = argv_copy[i];
+		arg_command = argv_copy[i + 1];
+
+		if (!is_number(argv_copy[i + 2])) {
+			TEST_FAIL("Argument: %s must be a number [%s]", argv_copy[i + 1],
+				  argv_copy[i + 2]);
+		}
+		arg_value = argv_copy[i + 2];
+
+		LOG_INF("Sending to: %s command: %s value: %s", arg_dev, arg_command, arg_value);
+		ret = cmd_send(3, arg_dev, arg_command, arg_value);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+int test_params_set(void)
+{
+	for (int i = 0; i < argc_copy; i++) {
+		if (strcmp(argv_copy[i], "tester") != 0) {
+			continue;
+		}
+
+		if (strcmp(argv_copy[i + 1], "dev") == 0) {
+			if (strcmp(argv_copy[i + 2], "CENTRAL_SOURCE") == 0) {
+				LOG_INF("Role set to Central/Source");
+				role = CENTRAL_SOURCE;
+			} else if (strcmp(argv_copy[i + 2], "PERIPHERAL_SINK") == 0) {
+				LOG_INF("Role set to Peripheral/Sink");
+				role = PERIPHERAL_SINK;
+			}
+		} else if (strcmp(argv_copy[i + 1], "acl_fail_limit_percent") == 0) {
+			if (is_number(argv_copy[i + 2])) {
+				fail_limit_percent[ACL] = strtol(argv_copy[i + 2], NULL, 10);
+				LOG_INF("Var set: %d", fail_limit_percent[ACL]);
+			} else {
+				LOG_ERR("Argument %s must be a number [%s]", argv_copy[i],
+					argv_copy[i + 2]);
+				return -EINVAL;
+			}
+		} else if (strcmp(argv_copy[i + 1], "iso_fail_limit_percent") == 0) {
+			if (is_number(argv_copy[i + 2])) {
+				fail_limit_percent[ISO] = strtol(argv_copy[i + 2], NULL, 10);
+				LOG_INF("Var set: %d", fail_limit_percent[ISO]);
+			} else {
+				LOG_ERR("Argument %s must be a number [%s]", argv_copy[i],
+					argv_copy[i + 2]);
+				return -EINVAL;
+			}
+		} else if (strcmp(argv_copy[i + 1], "acl_packets_limit_min") == 0) {
+			if (is_number(argv_copy[i + 2])) {
+				packets_limit_min[ACL] = strtol(argv_copy[i + 2], NULL, 10);
+				LOG_INF("Var set: %d", packets_limit_min[ACL]);
+			} else {
+				LOG_ERR("Argument: %s must be a number [%s]", argv_copy[i],
+					argv_copy[i + 2]);
+				return -EINVAL;
+			}
+		} else if (strcmp(argv_copy[i + 1], "iso_packets_limit_min") == 0) {
+			if (is_number(argv_copy[i + 2])) {
+				packets_limit_min[ISO] = strtol(argv_copy[i + 2], NULL, 10);
+				LOG_INF("Var set: %d", packets_limit_min[ISO]);
+			} else {
+				LOG_ERR("Argument: %s must be a number [%s]", argv_copy[i],
+					argv_copy[i + 2]);
+				return -EINVAL;
+			}
+		}
```

Comment:
Instead of having e.g. `iso_packets_limit_min` and `acl_packets_limit_min` have you consider having sub commands for `iso` and `acl` so you can do

```
acl packets_limit_min 50 fail_limit_percent 80 
iso packets_limit_min 70 fail_limit_percent 50
```

That way it will be easier to extend with more commands as each new argument will just require 1 additional parser, rather than 1 for ACL and 1 for ISO. 

If you want to expand with different values for CIS/BIS, that will also be easier

Reply:
That is a good point. I do want to handle this in a separate PR to get this merged and expand later.